dp/cloud/go/services/dp/storage/cbsd_manager.go

```diff
@@ -245,28 +254,33 @@ func (c *cbsdManagerInTransaction) updateCbsd(networkId string, id int64, data *
 	return err
 }
 
-func (c *cbsdManagerInTransaction) enodebdUpdateCbsd(data *DBCbsd) (*DBCbsd, error) {
+func (c *cbsdManagerInTransaction) enodebdUpdateCbsd(data *DBCbsd) error {
 	identifiers := []string{"cbsd_serial_number", "network_id"}
 	maskFields := append(identifiers, getEnodebdWritableFields()...)
 	mask := db.NewIncludeMask(maskFields...)
 	filters := sq.Eq{"cbsd_serial_number": data.CbsdSerialNumber}
 	cbsd, err := c.selectForUpdateIfCbsdExists(mask, filters)
+
 	if err != nil {
-		return nil, err
+		return err
 	}
-	if !ShouldENodeBDUpdate(cbsd, data) {
-		return cbsd, nil
+
+	columns := []string{"last_seen"}
+
+	if ShouldEnodebdUpdateInstallationParams(cbsd, data) {
+		cols := append(getEnodebdWritableFields(), "should_deregister")
```

Comment:
I'd rather do this `columns = append(getEnodebdWritableFields(), "should_deregister", "last_seen")` to avoid unnecessary copying (3 times as much)